### PR TITLE
Refactor rpc auth parsing

### PR DESCRIPTION
**Pull request by bvbfan**
It's just a code refactor to remove redundant code.

### CODE REVIEW

File: src/masternodes/mn_rpc.cpp [hunk 1]
```diff
@@ -108,10 +108,11 @@ CAmount EstimateMnCreationFee(int targetHeight) {
     return GetMnCreationFee(targetHeight);
 }
 
-std::vector<CTxIn> GetInputs(UniValue const& inputs) {
+static std::vector<CTxIn> ParseInputs(UniValue const& inputs) {
     std::vector<CTxIn> vin{};
-    for (unsigned int idx = 0; idx < inputs.size(); idx++) {
-        const UniValue& input = inputs[idx];
+    auto& array = inputs.get_array();
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:113]: Add const

File: src/masternodes/mn_rpc.cpp [hunk 2]
```diff
@@ -165,6 +169,27 @@ static CWallet* GetWallet(const JSONRPCRequest& request) {
     return pwallet;
 }
 
+static std::vector<CTxIn> GetMineAuthInputs(CWallet* const pwallet)
+{
+    CTxDestination destination;
+    for(auto& script : Params().GetConsensus().foundationMembers) {
+        if(IsMine(*pwallet, script) == ISMINE_SPENDABLE) {
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:176]: Can format the code a little bit, add blank after for and if

**Review by ShengguangXiao — APPROVED** [review]

**Comment by Bushstar** [comment]
Just a note to say that the Auto Auth PR below and this one will conflict. Auto Auth renames and changes some of the functions update in this PR. If we are to merge the Auto Auth one I think it needs to be done first and then update this PR against the new auth structure.

**Review by Bushstar — COMMENTED** [review]
Just a note to say that the Auto Auth PR below and this one will conflict. Auto Auth renames and changes some of the functions update in this PR. If we are to merge the Auto Auth one I think it needs to be done first and then update this PR against the new auth structure.

**Comment by bvbfan** [comment]
Sure, if other PR is accepted i'll rebase / merge.

**Comment by bvbfan** [comment]
#105 was merged closing for now.

**Pull request closed without merging**